File: frontend/src/api/activity.ts
import { useQuery } from '@tanstack/react-query'
import api from './client'

export interface Activity {
  id: string
  board_id: string
  user_id: string
  action: string
  entity_type: string
  entity_id: string
  metadata?: unknown
  created_at: string
  user?: {
    id: string
    display_name: string
    username: string
  }
}

export function useBoardActivity(boardId: string | null, limit = 50) {
  return useQuery({
    queryKey: ['activity', boardId],
    queryFn: async (): Promise<Activity[]> => {
      const res = await api.get(`/boards/${boardId}/activity?limit=${limit}`)
      return res.data || []
    },
    enabled: !!boardId,
    refetchInterval: false,
  })
}

File: backend/internal/models/activity.go
package models

import (
	"time"

	"github.com/jackc/pgx/v5/pgtype"
)

type Activity struct {
	ID         pgtype.UUID `json:"id"`
	BoardID    pgtype.UUID `json:"board_id"`
	UserID     pgtype.UUID `json:"user_id"`
	Action     string      `json:"action"`
	EntityType string      `json:"entity_type"`
	EntityID   pgtype.UUID `json:"entity_id"`
	Metadata   []byte      `json:"metadata,omitempty"`
	CreatedAt  time.Time   `json:"created_at"`
	User       *User       `json:"user,omitempty"`
}
